ovirtlib4/system_service.py

```diff
@@ -64,8 +70,94 @@ def _get_collection_entity(self):
         """
         return CollectionEntity(connection=self.connection)
 
-    def get(self, *args, **kwargs):
-        """ Same as list() """
+    def _process_sample(self, wait_for, wait_method, sample):
+        """
+        Handle a result sample according to the given 'wait_for' and 'wait_method'
+
+        Args:
+            wait_for (object): see run_sampler() docstring
+            wait_method (str): see run_sampler() docstring
+            sample (list): List of SDK entities, returned by get()
+
+        Returns:
+            list: If 'str' or 'bool' return the given sample if it match,
+                if 'str' return the filtered list if it match,
+                if no match return an empty list
+        """
+        if type(wait_for) == bool:
+            if (sample is not []) == wait_for:
+                return sample
+
+        if callable(wait_for):
+            results = eval("wait_for(sample)")
+            if results:
+                return sample
+
+        if type(wait_for) == str:
+            results = []
+            true_objects = []
+            for obj in sample:
+                result = eval(f"obj.{wait_for}")
+                results.append(result)
```

Comment:
maybe it will be better to do it with list comprehension like:
results = [ eval(f"obj.{wait_for}") for obj in sample ]
true_objects = [ result for result in results if result ]

Reply:
'results' hold the list of True, False 
while 'true_objects' hold the objects, 
To support what you suggested I need to go over the list twice
and on the first time save the result and the object in a tuple or dictionary,
and on the second loop save the objects according to the result field, so I don't see any benefit from that

Reply:
yep I missed it